core/src/main/resources/error/error-classes.json

```diff
@@ -1277,6 +1277,11 @@
           "A correlated outer name reference within a subquery expression body was not found in the enclosing query: <value>"
         ]
       },
+      "CORRELATED_REFERENCE" : {
+        "message" : [
+          "Expressions referencing the outer query are not supported outside of WHERE/HAVING clauses<treeNode>"
```

Comment:
Let's don't confuse users by `treeNode`, and show something more useful like problematic SQL expr.

Reply:
Agreed, let's show the expression. (having the expression in the context would also be acceptable). No treenode

Reply:
@MaxGekk May I ask if there is any way to extract the SQL expr from LogicalPlan ??

I'm trying to switch `planToString(p)` to SQL expression but still not sure how.

Reply:
Yes, just fixed it to show SQL expression!